diff --git a/engine/cdrs.go b/engine/cdrs.go
index 7feab366f..1b4e8f4f8 100644
--- a/engine/cdrs.go
+++ b/engine/cdrs.go
@@ -360,6 +360,12 @@ func (self *CdrServer) deriveCdrs(cdr *CDR) (drvdCDRs []*CDR, err error) {
 				break
 			}
 		}
+
+		// Carrier generates CDR only if calculateCost is enabled
+		if dc.RunID == "carrier" && cdr.ExtraFields["calculateCost"] != "1" {
+			continue
+		}
+
 		if !matchingAllFilters { // Do not process the derived charger further if not all filters were matched
 			continue
 		}
diff --git a/engine/responder.go b/engine/responder.go
index 4536f27ec..5243f358b 100644
--- a/engine/responder.go
+++ b/engine/responder.go
@@ -367,6 +367,12 @@ func (rs *Responder) GetDerivedMaxSessionTime(ev *CDR, reply *float64) (err erro
 				break
 			}
 		}
+
+		// Carrier does not limit MaxSessionTime
+		if dc.RunID == "carrier" {
+			continue
+		}
+
 		if !matchingAllFilters { // Do not process the derived charger further if not all filters were matched
 			continue
 		}
