diff --git a/engine/storage_redis.go b/engine/storage_redis.go
index 6e534466b..df8b3bffc 100755
--- a/engine/storage_redis.go
+++ b/engine/storage_redis.go
@@ -715,11 +715,21 @@ func (rs *RedisStorage) SetAccount(ub *Account) (err error) {
 	if len(ub.BalanceMap) == 0 {
 		if ac, err := rs.GetAccount(ub.ID); err == nil && !ac.allBalancesExpired() {
 			if len(ac.ActionTriggers) == 0 {
+				// Current account has no action trigger, override with new account's ones
 				ac.ActionTriggers = ub.ActionTriggers
+			} else if len(ub.ActionTriggers) == 0 {
+				// Current account has action triggers but new account doesn't, empty action triggers
+				ac.ActionTriggers = nil
 			}
-			if len(ub.UnitCounters) == 0 {
+
+			if len(ac.UnitCounters) == 0 {
+				// Current account has no unit counter, override with new account's ones
 				ac.UnitCounters = ub.UnitCounters
+			} else if len(ub.ActionTriggers) == 0 {
+				// Current account has unit counters but new account doesn't, empty unit counters
+				ac.UnitCounters = nil
 			}
+
 			ac.AllowNegative = ub.AllowNegative
 			ub = ac
 		}
